core/src/main/resources/error/error-classes.json

```diff
@@ -89,17 +131,40 @@
     "message" : [ "The second argument of '%s' function needs to be an integer." ],
     "sqlState" : "22023"
   },
+  "SHOULD_NOT_CALL_ALIAS_DO_GEN_CODE" : {
+    "message" : [ "Alias.doGenCode should not be called." ]
+  },
+  "SUM_OF_DECIMAL_OVERFLOW" : {
+    "message" : [ "Overflow in sum of decimals." ],
+    "sqlState" : "22003"
+  },
   "UNABLE_TO_ACQUIRE_MEMORY" : {
     "message" : [ "Unable to acquire %s bytes of memory, got %s" ]
   },
+  "UNEXPECTED_TYPE" : {
+    "message" : [ "Unexpected type %s" ],
+    "sqlState" : "42000"
+  },
   "UNRECOGNIZED_SQL_TYPE" : {
     "message" : [ "Unrecognized SQL type %s" ],
     "sqlState" : "42000"
   },
+  "UNSUPPORTED_DATATYPE" : {
+    "message" : [ "dataType" ],
+    "sqlState" : "0A000"
+  },
+  "UNSUPPORTED_INPUT_TYPE" : {
+    "message" : [ "Unsupported input type %s" ],
+    "sqlState" : "0A000"
+  },
   "UNSUPPORTED_LITERAL_TYPE" : {
     "message" : [ "Unsupported literal type %s %s" ],
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_ORDERED_OPERATIONS" : {
```

Comment:
This should be more specific, like `ORDERED_OPERATIONS_UNSUPPORTED_BY_DATA_TYPE`